BookStoreAPI/Interfaces/IBookService.cs

```diff
@@ -0,0 +1,15 @@
+﻿using BookStoreAPI.Models;
+using Microsoft.AspNetCore.JsonPatch;
+
+namespace BookStoreAPI.Interfaces;
+
+public interface IBookService
```

Comment:
Można zastosować tradycyjne podejście oparte na serwisach, ale łatwiej jest zapanować nad projektem, który ma rozdzielony kod na przypadki użycia takie które modyfikują stan aplikacji i takie które są read only. Można to zrobić przy pomocy CQRS o który można przeczytać więcej tutaj. https://cezarywalenciuk.pl/blog/programing/mediatr-cqrs-i-wzorzec-projektowy-mediator-w-aspnet-core Jeśli chcesz dalej rozwijać tę aplikację to warto to sprawdzić.